components/dueling-picklist/private/option.jsx

```diff
@@ -0,0 +1,175 @@
+		disabled: PropTypes.bool,
+	};
+
+	static defaultProps = {
+		tabIndex: null,
+		disabled: false,
+	};
+
+	constructor (props) {
+		super(props);
+		this.id = `dueling-picklist-option-${props.option.item.id}`;
+	}
+
+	handleClick = (event) => {
+		event.stopPropagation();
+		event.preventDefault();
+
+		const { shiftKey, metaKey, ctrlKey } = event;
+
+		const selectRange = shiftKey;
+		const selectAdditionalItem = metaKey || ctrlKey;
+
+		const { onSelect, option } = this.props;
+
+		onSelect(option.item, selectRange, selectAdditionalItem);
+	};
+
+	handleFocus = (event) => {
+		event.preventDefault();
+		event.stopPropagation();
+		const { onSelect, option } = this.props;
+
+		onSelect(option.item, false, false, true);
+	};
+
+	render () {
+		const {
+			option,
+			tabIndex,
+			connectDragSource,
+			connectDropTarget,
+			isDragging,
+			isDraggingWithGroup,
+			dragAndDropWithArrowKeys,
+			optionRef,
+			dragAndDropEnabled,
+			disabled,
+		} = this.props;
+		const { selected, item } = option;
+
+		const events = disabled
+			? {}
+			: {
+				onClick: this.handleClick,
+				onFocus: this.handleFocus,
+			};
+
+		const result = (
+			<li role="presentation" className="slds-listbox__item" id={this.id}>
+				<div
+					className={classNames(
+						'slds-listbox__option slds-listbox__option_plain slds-media slds-media_small slds-media_inline',
+						{
+							'slds-is-selected': !disabled && selected,
+							'slds-is-grabbed':
+								!disabled && dragAndDropWithArrowKeys && selected,
+						}
+					)}
+					aria-selected={!disabled && (selected || isDragging)}
+					role="option"
+					tabIndex={tabIndex}
+					ref={optionRef}
+					{...events}
+				>
+					<span className="slds-media__body">
+						<span className="slds-truncate" title={item.label}>
+							{item.label}
+						</span>
+					</span>
+					{this.renderLock()}
+				</div>
+			</li>
+		);
+
+		return !disabled && dragAndDropEnabled
+			? connectDragSource(connectDropTarget(result))
+			: result;
+	}
+
+	renderLock () {
+		const { option, assistiveText } = this.props;
+
+		if (!option.item.isLocked) {
+			return null;
+		}
+
+		return (
+			<span className="slds-media__figure slds-media__figure_reverse">
+				<span className="slds-icon_container" title={assistiveText.itemLocked}>
+					<Icon category="utility" name="lock" size="xx-small" />
+					<span className="slds-assistive-text">
+						{': '}
+						{assistiveText.lockedItemCannotBeMoved}
+					</span>
+				</span>
+			</span>
+		);
+	}
+}
+
+const optionSource = {
+	beginDrag (props) {
+		props.beginDrag();
+		return {
+			option: props.option,
+		};
+	},
+	endDrag (props) {
+		props.endDrag();
+	},
+};
+
+const optionSourceCollector = (connect, monitor) => ({
+	connectDragSource: connect.dragSource(),
+	isDragging: monitor.isDragging(),
+});
+
+const optionTarget = {
+	drop (props, monitor, component) {
+		const hoverBoundingRect = findDOMNode(component).getBoundingClientRect();
```

Comment:
I'm getting this eslint warning:
```
error  Do not use findDOMNode                                    react/no-find-dom-node
```

@interactivellama, do you know of an alternative approach for detecting Drag and Drop boundaries?